src/app/+search-page/search-service/search.service.ts

```diff
@@ -100,26 +102,7 @@ export class SearchService implements OnDestroy {
   }
 
   search(query: string, scopeId?: string, searchOptions?: SearchOptions): Observable<RemoteData<Array<SearchResult<DSpaceObject>>>> {
-    this.searchOptions = this.searchOptions;
-    let self = `https://dspace7.4science.it/dspace-spring-rest/api/search?query=${query}`;
-    if (hasValue(scopeId)) {
-      self += `&scope=${scopeId}`;
-    }
-    if (isNotEmpty(searchOptions) && hasValue(searchOptions.pagination.currentPage)) {
-      self += `&page=${searchOptions.pagination.currentPage}`;
-    }
-    if (isNotEmpty(searchOptions) && hasValue(searchOptions.pagination.pageSize)) {
-      self += `&pageSize=${searchOptions.pagination.pageSize}`;
-    }
-    if (isNotEmpty(searchOptions) && hasValue(searchOptions.sort.direction)) {
-      self += `&sortDirection=${searchOptions.sort.direction}`;
-    }
-    if (isNotEmpty(searchOptions) && hasValue(searchOptions.sort.field)) {
-      self += `&sortField=${searchOptions.sort.field}`;
-    }
-
-    const errorMessage = undefined;
-    const statusCode = '200';
+    const error = new RemoteDataError('200', undefined);
```

Comment:
The name `RemoteDataError` is a bit confusing, considering this seems to be returning a 200 response code (which obviously is not an error).  Should this be called something like `RemoteDataResponse`?

Reply:
It should be an error. This is the result of a find-and-replace. Instead of creating an error object with a status code 200 and a message of undefined, the error object should just be undefined. Thanks for spotting that, I'll fix it.